app/assets/scripts/components/upload-form.js

```diff
@@ -11,27 +11,27 @@ import SuccessModal from './success-modal';
 
 var UploadForm = React.createClass({
   displayName: 'UploadForm',
-  propTypes: {
-    errors: React.PropTypes.array,
-    showModal: React.PropTypes.bool
-  },
 
   getInitialState: function () {
     return {
+      metadata: {},
       errors: [],
+      formFile: 'Choose File to Upload',
       showModal: false
     };
   },
 
-  logOutput: function (failures) {
+  logOutput: function (failures, metadata) {
     let errorText = '';
     failures.forEach((failure) => {
       errorText += `${failure}\n`;
     });
     this.setState({
       errors: errorText,
```

Comment:
You should not store rendered values in the state. In this case you're storing a string with all the errors. What happens if we want to render the errors differently? You should be storing an array  with the info you need, and do the actual render when they're to be displayed.